### DISCUSSION TITLE
CupertinoSegmentedButton places segments not in listed order

### DISCUSSION THREAD

**Discussion by roob1e** [flet-dev/flet#4401] [Q&A]
### Duplicate Check

- [x] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the bug

CuperinoSegmentedButton's controls are assigned with elements of a certain list:
```python
controls = [ft.Text(subject) for subject in request]
```
But when I'm adding this class to my app, elements in button end up mixed up.
![image](https://github.com/user-attachments/assets/3da55ea4-52a3-4674-9829-c6cf83a797ba)

### Code sample

<details open><summary>Code</summary>

```python
import flet as ft

def main(page: ft.Page):
    page.window.width = 1920
    page.window.height = 1080
    page.window.maximized = True
    page.window.resizable = False
    page.theme_mode = "DARK"
    
    request = list({"1", "2", "3", "4"})
    
    selection_bar = ft.CupertinoSegmentedButton(
            selected_color = ft.Theme.primary_color_light,
            padding = ft.padding.all(20),
            width = 200,
            unselected_color = ft.colors.GREY_900,
            on_change = lambda e: print(e),
            controls = [ft.Text(subject) for subject in request]
    )
    
    c1 = ft.Container(
        content = selection_bar,
        alignment = ft.alignment.top_center
    )
    
    page.add(c1)
    page.update()

ft.app(target = main)      
```

</details>

### To reproduce

1. Run the code;
2. Notice the difference between list sequence and final sequence.

### Expected behavior

The order of the list elements matches the placement of the elements in the button.

### Screenshots / Videos

<details open>
<summary>Captures</summary>

![image](https://github.com/user-attachments/assets/e90287d9-1b3e-4004-92c7-515a64907a4c)
![image](https://github.com/user-attachments/assets/338f4932-1949-4b83-a39a-bd96a29f892e)
![image](https://github.com/user-attachments/assets/cc813890-83b9-4c99-8e72-aa057842c7fb)

</details>

### Operating System

Windows

### Operating system details

11

### Flet version

0.24.1

### Regression

I'm not sure / I don't know

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

**Comment by ndonkoHenri** [comment]
Python is the cause, not Flet.
That's how python sets function (they [dont't](https://stackoverflow.com/a/61467874) keep the provided order).

Try printing your `request` list somewhere in your code and you will notice what I mean.
